src/form/input/Input.tsx

```diff
@@ -75,6 +81,11 @@ function Input(props: InputProps) {
     "input--has-error": hasError
   });
   const isNumberInput = type === "number";
+  let finalValue = value;
+
+  if (isNumberInput && value && shouldFormatToLocaleString) {
+    finalValue = numberToString(parseNumber(String(value)), maxFractionDigits, locale);
```

Comment:
Here we will FORMAT the number into the localized string representation with "formatNumber" function